src/plugins/intel_gpu/src/plugin/infer_request_legacy.cpp

```diff
@@ -543,13 +543,11 @@ void InferRequestLegacy::SetGraph(std::shared_ptr<Graph> graph) {
 
 void InferRequestLegacy::SetBatch(int new_batch) {
     OV_ITT_SCOPED_TASK(itt::domains::intel_gpu_plugin, "InferRequestLegacy::SetBatch");
-    if (m_graph->GetMaxDynamicBatchSize() < 0)
-        IE_THROW() << "Dynamic batch is not enabled.";
 
-     if (new_batch < 1 || static_cast<size_t>(new_batch) > m_graph->GetMaxDynamicBatchSize()) {
-        IE_THROW() << "Invalid dynamic batch size " << new_batch <<
-            " for this request. Got: " << new_batch << ". Expected value in range [1;" << m_graph->GetMaxDynamicBatchSize() << "]";
-    }
+    OPENVINO_ASSERT(new_batch != 0 && static_cast<size_t>(new_batch) <= m_graph->GetMaxDynamicBatchSize(),
```

Comment:
As I can see, SetBatch has default new_batch=-1, is it your intention to cover this case with static_cast<size_t>(new_batch) casting?